### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by CyrusVonNox-01** [swfans/syndwarsfx#213] [Q&A]
What is the command prompt for when you start the game? I'm a little suspicious of applications that just open a command prompt like that. I'm sure it's completely legitimate, I've been following this project since last year, but it would still would be nice to know.

**Comment by mefistotelis** [comment]
We don't have that much use of it. It shows some logs from libraries which send the logging to stdout by default (but most of the logs are in `error.log` instead). And the main reason it was re-enabled with SDL2 is the ability to show help:
https://github.com/swfans/swars/issues/194

EDIT:
Also worth mentioning this is a windows-only uncertainty. Linux users know each application has a console, and Linux provides users several ways to hide it if it isn't necessary.

**Comment by geist22** [comment]
It would be great if we had something like a `-noconsole` option to not show the additional window, as it's usually just getting in the way for the user. Or perhaps even better, disable it by default and require an option to enable it if desired.

**Reply by mefistotelis** [reply]
Not sure if that's possible. Under Windows, marking an app as GUI only is done in Portable Executable headers, not determined by the app code.

**Reply by geist22** [reply]
Not sure how related this is, but [DOSBox Staging](https://github.com/dosbox-staging/dosbox-staging) does have this `-noconsole` option. The cmd window is still briefly visible at the start, then closes before the main window appears.

**Reply by mefistotelis** [reply]
I see. Yeah, that looks totally doable:

https://github.com/dosbox-staging/dosbox-staging/blob/690e5874466ee5cb599b5e91410f264f21687828/src/gui/sdlmain.cpp#L4896
